src/statscollector/StatsCollector.ts

```diff
@@ -618,4 +621,48 @@ export default class StatsCollector implements RedundantAudioRecoveryMetricsObse
 
     this.lastRedRecoveryMetricReportConsumedTimestampMs = this.redRecoveryMetricReport.currentTimestampMs;
   }
+
+  /**
+   * Receive video codec degradation event due to high encode CPU usage
+   * from MonitorTask and increment counter
+   */
+  videoCodecDegradationHighEncodeCpuDidReceive(): void {
+    this.videoCodecDegradationHighEncodeCpuCount += 1;
+  }
+
+  /**
+   * Receive video codec degradation event due to hardware encoder failure
+   * from MonitorTask and increment counter
+   */
+  videoCodecDegradationEncodeFailureDidReceive(): void {
+    this.videoCodecDegradationEncodeFailureCount += 1;
+  }
+
+  private getVideoUpstreamSsrc(): number | null {
```

Comment:
nit: Isn't this the same function as in `ClientMetricReport`? Can we just modify the one in `getVideoUpstreamSsrc` to make it reusable here?

Reply:
Removed the function in `StatsCollector`